alpaca/common/exceptions.py

```diff
@@ -8,9 +8,26 @@ class APIError(Exception):
     """
 
     def __init__(self, error, http_error=None):
-        super().__init__(error)
+        detailed_error = {
+            "payload": error,
+        }
+
+        if http_error is not None:
+            if hasattr(http_error, "response"):
+                detailed_error["status_code"] = http_error.response.status_code
+                detailed_error["reason"] = http_error.response.reason
+            if hasattr(http_error, "request"):
+                detailed_error["method"] = http_error.request.method
+                detailed_error["url"] = http_error.request.url
+            # add tips for auth key error
+            if detailed_error["status_code"] in [401, 403]:
```

Comment:
Do you need 403 here? If the URL is incorrect, you should check only 401?

Reply:
If I connect to live (sandbox=False) with a key for sandbox, it returns 403 as below. Therefore, it might be worth to include 403 ?

```
from alpaca.broker.client import BrokerClient
client = BrokerClient(api_key=BROKER_API_KEY, secret_key=BROKER_API_SECRET, sandbox=False)
client.list_accounts()
```

> APIError: {'payload': '{"message": "forbidden."}\n', 'status_code': 403, 'reason': 'Forbidden', 'method': 'GET', 'url': 'https://broker-api.alpaca.markets/v1/accounts', 'tips': 'please check your API key and environment (paper/sandbox/live)'}

Reply:
If should be one of those. 403 is sad because API returns it for other cases as well such as not enough shares to sell. Downside of this approach is that the new error could be confusing in those cases unless you are really specific to the particular api auth error case

Reply:
Understood. Thank you for the info. In that case, agree, 401 only would be nicer. User still can get a hint by url strings.

Reply:
My question is, does api even return 401?

Reply:
Yes. API returns 401 when paper=False and use a key for paper as I wrote the first case in description.

Reply:
returns tips message only for 401. Addressed https://github.com/alpacahq/alpaca-py/pull/391/commits/b7f027f12d1c4876835a5352241a3825730a5688